File: solution_set1/2264. Largest 3-Same-Digit Number in String.py
##ss
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        
        comp = ""
        ans = -1
        
        for x in range(len(num) - 2):
            
            thisnums = num[x:x+3]
            
            if (thisnums[0] == thisnums[1] == thisnums[2]) and int(thisnums) > ans:
                ans = int(thisnums)
                comp = thisnums
                
        return comp
        

File: cpp_solutions/2264. Largest 3-Same-Digit Number in String.cpp
//ss
class Solution {
public:
    string largestGoodInteger(string num) {
        int size = num.length();
        int ans = -1;
        
        for(int x =0; x<size-3+1;x++){
            if (num.at(x) == num.at(x+1) && num.at(x+1) == num.at(x+2)){
                //cout << stoi(num.substr(x,3)) << "\n";
                ans = max(ans, stoi(num.substr(x,3)));
                //return num.substr(x,x+2);
            }
        }
        
        if (ans == -1){return "";}
        else if(ans == 0) {return "000";}
        else {return to_string(ans);}
        
    }
};
